cmd/gen-go-sample/testdata/sample_unary.want

```diff
@@ -36,6 +38,25 @@ func sampleUnaryMethod(theX int64, theB string, theFoo string) error {
 	// theX := 42
 	// theB := "foobar"
 	// theFoo := "myfoo"
+	// fileArg := "path/to/local/file/arg.txt"
+	file, err := os.Open("path/to/local/file/default.txt")
+	if err != nil {
+		log.Fatalf("Failed to read file: %v", err)
+	}
+	defer file.Close()
+	fileDefaultContent, err := ioutil.ReadAll(file)
+	if err != nil {
+		log.Fatalf("Failed to read file: %v", err)
+	}
+	file, err := os.Open(fileArg)
```

Comment:
Previous comment WRT file variable naming. This line will produce a `no new variables on left side of :=` error I believe.

Reply:
Agreed. We should either have logic so that only the first time we use the `:=`, or use another structure like
```
var file io.Reader
var err error
file, err = os.Open(...)
...
file, err = os.Open(...)
...
```

Reply:
(Full disclaimer from Go newbee)

> We should either have logic so that only the first time we use the :=
i.e. something like this:

```
file, err := os.Open("the/first/file.txt")
...
file, err = os.Open("another/file.txt")
...
```

Reply:
Yes that would address @vchudnov-g 's suggestion.

Personally i'd vote for the latter suggestion, explicit variable declaration 100% of the time. I think it would make your logic easier (don't have to save any state to indicate whether or not you've seen a bytes field already). And it's still solid Go.

Very minor bonus, but it's more self-documenting b.c the `io.Reader` is declared explicitly implying something other than an `os.File` could fulfill the example as well.

Reply:
Sounds good. I'll change the code to use this approach.

Reply:
The one caveat with that second suggestion: if there is only one file that needs to be opened, then IIUC it's probably more idiomatic to skip the `var` and use the `:=`. But if we have more than one file, so that we need to reuse the vars, using `var` is probably preferred (eg for the reason's @noahdietz suggested)

Reply:
See my suggestion on the code that generates this. I think we can use `ioutil.ReadFile` to avoid giving file handle a name altogether.

Reply:
Agreed. We should use `ioutil.ReadFile`

Reply:
Ohhh, I thought the suggestion was just about to use `fmt.Fprintf`. I didn't realize using `ioutil.ReadFIle` will get rid of all the troubles!